server/src/actors/events.rs

```diff
@@ -550,6 +557,47 @@ impl Handler<HandleEvent> for EventManager {
             .map_err(move |error, _, _| {
                 log::warn!("error processing event {}: {}", event_id, LogError(&error));
                 metric!(counter("event.rejected") += 1);
+                let outcome_params: Option<Outcome> = match &error {
+                    | ProcessingError::SerializeFailed(_)
+                    | ProcessingError::ScheduleFailed(_)
+                    | ProcessingError::PiiFailed(_)
+                    | ProcessingError::Timeout
+                    | ProcessingError::Shutdown => {
+                        Some(Outcome::Invalid(DiscardReason::Internal))
+                    }
+
+                    ProcessingError::InvalidJson(_) => {
+                        Some(Outcome::Invalid(DiscardReason::InvalidPayloadJsonError))
+                    }
+
+                    #[cfg(feature = "processing")]
+                    ProcessingError::StoreFailed(_store_error) => {
+                        Some(Outcome::Invalid(DiscardReason::Internal))
+                    }
+                    ProcessingError::EventRejected(outcome_reason) => {
+                        Some(Outcome::Invalid(*outcome_reason))
```

Comment:
_nit_: Match directly on `error`. `OutcomeReason` implements `Copy`, so you do not need the implicit ref here. Same applies to `FilterStatKey` below. You will need to match `RateLimited(_timeout, ref reason)` below, though.

Reply:
ok, done